src/construction.cpp

```diff
@@ -820,17 +827,35 @@ void place_construction( const std::string &desc )
         cons.front()->explain_failure( pnt );
         return;
     }
-
+    std::list<item> used;
     const construction &con = *valid.find( pnt )->second;
-    g->u.assign_activity( activity_id( "ACT_BUILD" ), con.adjusted_time(), con.id );
+    // create the partial construction struct
+    partial_con pc;
+    pc.id = con.id;
+    pc.counter = 0;
+    // Set the trap that has the examine function
+    g->m.trap_set( pnt, tr_unfinished_construction );
+    // Use up the components
+    for( const auto &it : con.requirements->get_components() ) {
+        std::list<item> tmp = g->u.consume_items( it, 1, is_crafting_component );
+        used.splice( used.end(), tmp );
+    }
+    pc.components = used;
+    g->m.partial_con_set( pnt, pc );
+    for( const auto &it : con.requirements->get_tools() ) {
+        g->u.consume_tools( it );
+    }
+
+    g->u.assign_activity( activity_id( "ACT_BUILD" ) );
+    g->u.activity.index = con.id;
```

Comment:
This is already stored in the partial construction, so I think it would be cleaner to just get it from there.